src/site/content/en/blog/usb/index.md

```diff
@@ -0,0 +1,369 @@
+promise is fulfilled with a `result` object containing a [DataView] `data` that
+has to be parsed appropriately.
+
+For those who are familiar with USB, all of this should look pretty familiar.
+
+### I want more
+
+The WebUSB API lets you interact with the all USB transfer/endpoint types:
+
+- CONTROL transfers, used to send or receive configuration or command
+  parameters to a USB device are handled with `controlTransferIn(setup,
+  length)` and `controlTransferOut(setup, data)`.
+- INTERRUPT transfers, used for a small amount of time sensitive data are
+  handled with the same methods as BULK transfers with
+  `transferIn(endpointNumber, length)` and `transferOut(endpointNumber, data)`.
+- ISOCHRONOUS transfers, used for streams of data like video and sound are
+  handled with `isochronousTransferIn(endpointNumber, packetLengths)` and
+ `isochronousTransferOut(endpointNumber, data, packetLengths)`.
+- BULK transfers, used to transfer a large amount of non-time-sensitive data in
+  a reliable way are handled with `transferIn(endpointNumber, length)` and
+ `transferOut(endpointNumber, data)`.
+
+You may also want to have a look at Mike Tsao's [WebLight project] which
+provides a ground-up example of building a USB-controlled LED device designed
+for the WebUSB API (not using an Arduino here). You'll find hardware, software,
+and firmware.
+
+## Tips
+
+Debugging USB in Chrome is easier with the internal page `chrome://device-log`
+where you can see all USB device related events in one single place.
+
+<figure class="w-figure">
+  <img src="./web-usb-device-log-page.png" class="w-screenshot"  alt="Screenshot of the device log page to debug WebUSB in Chrome">
+  <figcaption class="w-figcaption">Device log page in Chrome for debugging the WebUSB API.</figcaption>
+</figure>
+
+The internal page `chrome://usb-internals` also comes in handy and allows you
+to simulate connection connection and disconnection of virtual WebUSB devices.
+This is be useful for doing UI testing without the need for real hardware.
+
+<figure class="w-figure">
+  <img src="./web-usb-internals-page.png" class="w-screenshot" alt="Screenshot of the internal page to debug WebUSB in Chrome">
+  <figcaption class="w-figcaption">Internal page in Chrome for debugging the WebUSB API.</figcaption>
+</figure>
+
+On most Linux systems, USB devices are mapped with read-only permissions by
+default. To allow Chrome to open a USB device, you will need to add a new [udev
+rule]. Create a file at `/etc/udev/rules.d/50-yourdevicename.rules` with the
+following content:
+
+```vim
+SUBSYSTEM=="usb", ATTR{idVendor}=="[yourdevicevendor]", MODE="0664", GROUP="plugdev"
+```
+
+where `[yourdevicevendor]` is `2341` if your device is an Arduino for instance.
+`ATTR{idProduct}` can also be added for a more specific rule. Make sure your
+`user` is a member of the `plugdev` group.  Then, just reconnect your device.
+
+## What's next
```

Comment:
Does this section need to be updated? I couldn't find any features for shared or service workers on Chrome Status. I did find this: https://www.chromestatus.com/features/5928209916887040

Reply:
Oh I've missed this. You're right.
I've removed this "What's next" section